public/app/features/dashboard/dashboard_model.ts

```diff
@@ -146,6 +152,18 @@ export class DashboardModel {
       list: _.map(this.templating.list, variable => (variable.getSaveModel ? variable.getSaveModel() : variable)),
     };
 
+    if (!saveVariables && copy.templating.list.length === this.originalTemplating.length) {
```

Comment:
Tests and verification of saving/not saving variables is behaving as expected is needed.